### ISSUE TITLE
insights: FE paging for insights

### ISSUE THREAD

**Issue by unclejustin** [sourcegraph/sourcegraph#27221]

**Comment by felixfbecker** [comment]
Alternative proposal: What if we got rid of the ability to resize and reorder and `react-grid-layout`? Order and size was never persisted anyway. If we ever wanted to add it (unlikely to be a priority anytime soon) we probably would have migrated to a different library anyway.

The only argument I heard from @Joelkw was that the resizing was helpful to _enlarge_ an insight in lieu of a single insights page, but that is now also possible through the single insights page. So could we easily get rid of that complexity?

It seems that would then make scrolling pagination trivial (using any off-the-shelf React scroll library).

**Comment by Joelkw** [comment]
@felixfbecker I like the proposal and creative angle, but I see enough users that are used to resizing the insights on dashboards I'm not sure that's the right direction. Rather than pagination, which would suggest some ordering that is more deterministic for multiple users (it'll be diff per user, though), can we just do a "load more" button after 40 insights or something? 

Thanks for flagging this @unclejustin .

**Comment by unclejustin** [comment]
Load more works for me and would solve the core problem. A+ suggestion @Joelkw 

If and when we get to this @AlicjaSuska would it be helpful for me to just build this _then_ tag you for review? Or would you prefer to design first?

**Comment by Joelkw** [comment]
@vovakulikov fair point, but we should set "load more" to be generous so this still works for most customers for a while, and most users will have fewer insights than will break the page / than the 300+ on k8s. See this dashboard – https://sourcegraph.looker.com/dashboards/199?Unique+Server+ID= – what if we set it at 50 insights, for example?

**Comment by unclejustin** [comment]
@Joelkw my thoughts exactly. I think _most_ customers would never even see this. I would back burner the search though until someone actually complains though. Our internal use case is very different from a customer's.
